src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractValidationServiceImpl.java

```diff
@@ -203,38 +210,60 @@ private boolean validateReportId(ConcurStandardAccountingExtractDetailLine line,
+            }
+            return validationResults.isValid();
+        } else {
+            LOG.debug("validateAccountingLine, found a cash advance line, no need to validate");
+            return true;
+        }
+    }
+
+    private boolean validateAccountingLineWithObjectCodeOverrides(ConcurStandardAccountingExtractDetailLine line, 
+            ConcurStandardAccountingExtractBatchReportData reportData, String overriddenObjectCode, String overriddenSubObjectCode) {
+        if (!getConcurStandardAccountingExtractCashAdvanceService().isCashAdvanceLine(line)) {
             logErrorsWithOriginalAccountingDetails(line);
     
-            ValidationResult overriddenValidationResults = buildValidationResult(buildOverriddenConcurAccountingInformation(line), true);
+            ValidationResult overriddenValidationResults = buildValidationResult(
+                    buildConcurAccountingInformation(line, overriddenObjectCode, overriddenSubObjectCode), true);
     
             if (overriddenValidationResults.isNotValid()) {
-                reportData.addValidationErrorFileLine(new ConcurBatchReportLineValidationErrorItem(line.getReportId(), line.getEmployeeId(), line.getEmployeeLastName(), line.getEmployeeFirstName(), line.getEmployeeMiddleInitital(), overriddenValidationResults.getMessages()));
+                reportData.addValidationErrorFileLine(new ConcurBatchReportLineValidationErrorItem(line.getReportId(), line.getEmployeeId(), 
+                        line.getEmployeeLastName(), line.getEmployeeFirstName(), line.getEmployeeMiddleInitital(), overriddenValidationResults.getMessages()));
             }
 
             return overriddenValidationResults.isValid();
         } else {
-            LOG.debug("validateAccountingLine, found a cash advance line, no need to validate");
+            LOG.debug("validateAccountingLineWithObjectCodeOverrides, found a cash advance line, no need to validate");
             return true;
         }
     }
-    
-    private ConcurAccountInfo buildOverriddenConcurAccountingInformation(ConcurStandardAccountingExtractDetailLine line) {
-        String overriddenObjectCode = getParameterService().getParameterValueAsString(CUKFSConstants.ParameterNamespaces.CONCUR, 
-                CUKFSParameterKeyConstants.ALL_COMPONENTS, ConcurParameterConstants.CONCUR_SAE_PDP_DEFAULT_OBJECT_CODE);
-        return buildConcurAccountingInformation(line, overriddenObjectCode, StringUtils.EMPTY);
-    }
 
     private void logErrorsWithOriginalAccountingDetails(ConcurStandardAccountingExtractDetailLine line) {
         ConcurAccountInfo accountingInformation = buildConcurAccountingInformation(
                 line, line.getJournalAccountCode(), line.getSubObjectCode());
```

Comment:
This line should be updated to call the new one-arg buildConcurAccountingInformation() method so that it logs errors for the correct sub-object code.